### PR TITLE
feat(asset): query builder for list assets

### PR DESCRIPTION
**Pull request by Catalin-web**
# Pull Request

## 🤨 Rationale

Task: [Add basic query builder: with workspace](https://ni.visualstudio.com/DevCentral/_backlogs/backlog/ASW%20SystemLink%20Asset%20Mgmt/Features/?workitem=2892128)

## 👩‍💻 Implementation

Added query builder for list assets

## 🧪 Testing

Modified & added tests+
list assets query builder

## ✅ Checklist

- [x] This PR has a title that follows the [commit message format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format).

### CODE REVIEW

**Review by MileaRobertStefan — APPROVED** [review]

**Comment by MileaRobertStefan** [comment]
<type>(<scope>): <short summary>
  │       │             │
  │       │             └─⫸ Summary in present tense. **Not capitalized**. No period at the end.
  │       │
  │       └─⫸ Commit Scope: If your change applies to a single plugin, put its short name
  │                          as the scope (e.g. tag, notebook, system).
  │
  └─⫸ Commit Type: build|ci|docs|feat|fix|perf|refactor|test

File: src/datasources/asset/components/editors/AssetDataSourceBase.ts [hunk 1]
```diff
@@ -11,4 +24,61 @@ export abstract class AssetDataSourceBase extends DataSourceBase<AssetQuery, Ass
   testDatasource(): Promise<TestDataSourceResponse> {
     throw new Error("Method not implemented.");
   }
+
+  public async loadDependencies(): Promise<void> {
```
Comment (MileaRobertStefan) [inline src/datasources/asset/components/editors/AssetDataSourceBase.ts:28]: nit: move this near the `loadSystems` and  `loadWorkspaces`.

File: src/datasources/asset/components/editors/list-assets/query-builder/AssetQueryBuilder.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        dataSource: [
+          ...(locationField.lookup?.dataSource || []),
+          ...systems.map(({ id, alias }) => ({ label: alias || id, value: id })),
+        ],
+      },
+    };
+  }, [systems]);
+
+  useEffect(() => {
+    if (areDependenciesLoaded) {
```
Comment (MileaRobertStefan) [inline src/datasources/asset/components/editors/list-assets/query-builder/AssetQueryBuilder.tsx:70]: nit: use a guard clause

**Review by kkerezsi — APPROVED** [review]

File: src/datasources/asset/components/editors/AssetDataSourceBase.ts [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+import { Workspace } from "../../../../core/types";
+import { metadataCacheTTL } from "../../../data-frame/constants";
 
 export abstract class AssetDataSourceBase extends DataSourceBase<AssetQuery, AssetDataSourceOptions> {
+  public areSystemsLoaded = false;
+  public areWorkspacesLoaded = false;
+
+  public error = '';
+
+  private readonly systemAliasCache: TTLCache<string, SystemMetadata> = new TTLCache<string, SystemMetadata>({ ttl: metadataCacheTTL });
```
[the remark this answers is not in the document]
Comment (Catalin-web) [reply src/datasources/asset/components/editors/AssetDataSourceBase.ts]: I think this is outdated

**Pull request merged**